programs/odbc-bridge/ODBCConnectionFactory.h

```diff
@@ -77,12 +97,15 @@ class ODBCConnectionFactory final : private boost::noncopyable
 
         try
         {
-            if (!connection || !connection->connected())
+            if (!connection || needReconnect(*connection))
```

Comment:
Maybe just catch 'no connection' exception during the user's query preparation/execution and reconnect / retry query?

Reply:
I doubt it is ok, because there are quite many places where connection is used and it is used not in a uniform way. Adding try catch with reconnect and retry in all those places will look bad..